app/services/status_update_service.rb

```diff
@@ -3,23 +3,32 @@ def initialize(reference:, status:, user: nil)
     @reference = reference
     @status = status
     @user = user
+    @delivery_attempt = find_delivery_attempt(reference)
```

Comment:
This class originally did look up the delivery attempt, but we removed it as part of speeding up this endpoint when we couldn't cope with the number of requests coming in from Notify, so this endpoint made only one query (the `UPDATE`(. This was before we added the database index so it may not be a problem anymore, but worth noting.

Reply:
@thomasleese yeah we have an email to update as well so I figured it was more prudent to grab email and delivery attempt together. Could probably at a push switch over to be just an update on each without the selects but lose the ability to do the preconditions. 

I realise I should always add a lock for update to it if I want to really stomp out the risk of conflicts.